File: src/main/java/org/itmo/isLab1/dragons/dto/DragonDto.java
package org.itmo.isLab1.dragons.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;

import org.itmo.isLab1.common.framework.dto.CrudDto;
import org.itmo.isLab1.coordinates.Coordinate;
import org.itmo.isLab1.dragoncaves.DragonCave;
import org.itmo.isLab1.dragonheads.DragonHead;
import org.itmo.isLab1.dragons.enums.DragonCharacter;
import org.itmo.isLab1.dragons.enums.DragonType;
import org.itmo.isLab1.people.Person;
import org.itmo.isLab1.people.enums.Color;

@Data
@EqualsAndHashCode(callSuper = true)
public class DragonDto extends CrudDto {
    private int id;
    private String name;
    private Coordinate coordinates;
    private DragonCave cave;
    private Person killer;
    private Integer age;
    private Color сolor;
    private DragonType type;
    private DragonCharacter character;
    private DragonHead head;
}

File: frontend/src/interfaces/dto/dragons/DragonDto.ts
import { CrudDto } from '../CrudDto';
import { Color } from '../../models/Colors';
import { DragonCharacter } from '../../models/DragonCharacter';
import { CoordinateDto } from '../coordinates/CoordinateDto';
import { DragonCaveDto } from '../dragoncaves/DragonCaveDto';
import { DragonHeadDto } from '../dragonheads/DragonHeadDto';
import { DragonType } from '../../models/DragonType';
import { PersonDto } from '../people/PersonDto';

export interface DragonDto extends CrudDto {
    name: string;
    coordinates: CoordinateDto;
    cave?: DragonCaveDto;
    killer?: PersonDto;
    age?: number;
    color?: Color;
    type?: DragonType;
    character: DragonCharacter;
    head?: DragonHeadDto;
}